src/nunit.analyzers/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzer.cs

```diff
@@ -164,11 +165,26 @@ private static void AnalyzeAttribute(SyntaxNodeAnalysisContext context)
                         syntaxNode.GetLocation().SourceSpan.Start,
                         symbol);
 
-                    if (attributeInfo.IsStringLiteral && sourceIsAccessible)
+                    if (attributeInfo.IsStringLiteral && sourceIsAccessible && stringConstant != null)

```

Comment:
Surprised to see this null check only here. Shouldn't it be somewhere higher?
Not sure that usages on lines 160 and 188 are supposed to accept null.

Reply:
You are right. I got an error that I could not palce stringConstant into `Dictionary<string, string>`, so I added the check without thinking much about it. Looking more closely, I think I'll move it up to 149, so that the condition becomes `syntaxNode == null || stringConstant == null`.